modules/internal/localeModels/gpu/LocaleModel.chpl

```diff
@@ -379,7 +379,16 @@ module LocaleModel {
     //- Implementation (private)
     //-
     proc setup() {
+      extern proc chpl_gpu_on_locale_model_finished_initializing() : void;
+
       helpSetupLocaleGPU(this, local_name, numSublocales, GPULocale);
+
+      // There are some allocations that occur in `helpSetupLocaleGPU` where we
+      // set the current sublocale to a GPU but we still want the allocations
+      // to actually be performed on the CPU. To do this we start the GPU
+      // runtime library in a mode where all allocations will be done on the
+      // host until this "finished_initializing" function is called.
+      chpl_gpu_on_locale_model_finished_initializing();
```

Comment:
I might have added this function, but "on" in the name doesn't read right to me?

Reply:
The original name was 'chpl_gpu_on_std_modules_finished_initializing', so in part I carried over the 'on' from there.

In the past I've used functions starting with the word "on" for things that are callbacks, follow the observer pattern, etc. so with that idiom in mind I think 'chpl_gpu_on_locale_model_finished_initializing' makes sense.

That said, this function only gets called once and it doesn't seem like we use 'on' in this way elsewhere in the runtime. I see the word 'hook' used for callbacks like 'chpl_memhook_malloc_pre'. What do you think about 'chpl_gpuhook_locale_model_initialized'?  If you that's too strange, I'll just directly go with 'chpl_gpu_locale_model_initialized'.

Reply:
> In the past I've used functions starting with the word "on" for things that are callbacks, follow the observer pattern, etc. so with that idiom in mind I think 'chpl_gpu_on_locale_model_finished_initializing' makes sense.

Ah, I see. I can read it like that for sure. I like replacing `finished_initializing` with just `initialized`, or even `inited`. For prefixes, I like `chpl_gpu` > `chpl_gpu_on` > `chpl_gpuhook`.

Reply:
Working on https://github.com/chapel-lang/chapel/pull/22172#discussion_r1188130456, I was thinking whether moving this call here is safe w.r.t. what's happening with `CHPL_GPU_DEBUG` outputs inside that function. Those `CHPL_GPU_DEBUG` calls must be done after the `ChapelGpuSupport` module is initialized. I suspect this is happening before that.

If this is a problem, I wonder whether this call can be made at the end of the `ChapelGpuSupport` module in a `coforall l in Locales do on l`? At a high-level that makes good sense to me to the extent that I would feel better about existence of this function as well. We'd be signaling the runtime that the crucial modules have finished initializing.

Reply:
I've moved the call into `ChapelGpuSupport` and renamed the function `chpl_gpu_support_module_finished_initializing`